modules/EvManager/main/RegisteredCommand.hpp

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Pionix GmbH and Contributors to EVerest
+
+#ifndef TMP_CLION_CLANG_TIDY_SIMCOMMAND_HPP
+#define TMP_CLION_CLANG_TIDY_SIMCOMMAND_HPP
+
+#include <cstddef>
+#include <memory>
+#include <mutex>
+#include <stdexcept>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+namespace module::main {
+
+template <typename Func> class RegisteredCommand;
+
+class RegisteredCommandBase {
+public:
+    RegisteredCommandBase() = default;
+    virtual ~RegisteredCommandBase(){};
+    RegisteredCommandBase(const RegisteredCommandBase&) = default;
+    RegisteredCommandBase& operator=(const RegisteredCommandBase&) = default;
+    RegisteredCommandBase(RegisteredCommandBase&&) = default;
+    RegisteredCommandBase& operator=(RegisteredCommandBase&&) = default;
```

Comment:
Honestly, I'm not sure what's best here. I was following https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-five. I'm not sure if this rule would apply here or not. What do you think?

Reply:
You have an abstract base class because of `virtual bool operator()(...) const = 0`, therefor you'll never be able to create an instance of `RegisteredCommandBase`.  If you can't create an instance, you also can't move/copy it. What you have, are derived instances of `RegisteredCommandBase`, these you can move/copy, if you want.  Usually you're referring to these derived instance via a `RegisteredCommandBase` pointer or reference.  Now you could of course dereference them via their parent type and call the copy/move operator - but this will potentially not do what you want (like when you have a non-virtual base class destructor), as long as you don't make the move/copy constructors of the base class virtual.  But again, as you're usually using only the `RegisteredCommandBase` pointer or reference, you don't have to move at all because your just moving the pointer and not the instance itself.

Reply:
That makes sense. Thanks!